### DISCUSSION TITLE
CS8034 warning when referencing Source Generator locally

### DISCUSSION THREAD

**Discussion by AydinE** [dotnet/roslyn#72937] [Q&A]
Hello,

The past few weeks I've been trying to develop a source generator to be used in our projects. I've now come to a point where the source generator works as expected however there is a CS8034 warning (Visual Studio only) that I just can't seem to get rid of.

The Source generator project itself is quite simple:
https://github.com/Lombiq/Helpful-Libraries/pull/251/files

As you can see, within the same solution the source generator is then referenced like so:
https://github.com/Lombiq/Helpful-Libraries/pull/251/files#diff-bc5ef2f21ac09705d622b943b9261ec0249e73a12bb7d18b7aa435cc6027e45fR27

Within this test project the source generator works as expected and creates the constants no problem, however we would like to get rid of this last pesky warning before merging this in.
![image](https://github.com/dotnet/roslyn/assets/1669059/61ffcfd7-c621-4fa6-a917-036016b380f3)

Some additional notes about this:
- Warning only seems to happen on a fresh clone of the solution.
- After a first build of the solution (and a restart of Visual Studio) the warning doesn't come back.
- Warning only seems to show up in Visual Studio (2022) - I personally use Rider and had never seen the warning until someone else pointed it out.
- Even specifically trying to suppress this warning with `<NoWarn>` doesn't seem to work 

Any insights on this would be greatly appreciated.

**Comment by huoyaoyuan** [comment]
The reason is quite straight: the IDE (VS) needs to load a built version of source generator to analyze all the code, but the built version doesn't exist in a fresh clone. It doesn't try to build one until you invoke the build command.
This is a known issue of IDE integration of source generator, since the first version. I don't know if there's any improvement planed.
Once you update code in source generator, you may need to build and restart VS to make integration works correctly.
